docs/bots-guide.md

```diff
@@ -0,0 +1,188 @@
+# Developing bots
+**This feature is still experimental**
+Bots are a new system in Zulip, allowing developers to react to messages.
+
+With bots, you *can*
+
+* intercept and view messages sent by users on Zulip
+* send out new messages
+
+With bots, you *cannot*
+
+* modify an intercepted message (you have to send a new message)
+* send messages in the name of other users
+* intercept private messages
+
+
+On this page you'll find:
+
+* A step-by-step [tutorial](#how-to-deploy-a-bot) on how to deploy a bot.
+* A step-by-step [tutorial](#how-to-develop-a-bot) on how to develop a bot.
+* A [documentation](#bot-api) of the bot API.
+* Common [problems](#common-problems) when developing/deploying bots and their solutions.
+
+Contributions to this guide are very welcome, so if you run into any
+issues following these instructions or come up with any tips or tools
+that help writing integration, please email
+zulip-devel@googlegroups.com, open an issue, or submit a pull request
+to share your ideas!
+
+## How to deploy a bot
+This guide will show you how to deploy a bot on your running Zulip server. It presumes that you already have a fully implemented `<my-bot>.py` bot and now want to try it out.
+
+1. Copy your bot `<my-bot>.py` to `~/zulip/contrib_bots/lib/`.
+* This is the place where all Zulip bots are stored.
+* You can also test out already existing bots in this directory.
+2. Run your Zulip server. Bots can only be deployed on running systems.
+3.  Register a new bot at your Zulip server's web interface.
+* Navigate to *Settings* -> *Your Bots* -> *Add a New Bot*, fill out the form and click on *Create Bot*.
+*  A new bot should appear in the *Your Bots* panel.
+4. Add the bot's configuration file on your Zulip server.
+* In the *Your Bots* panel, click on the green icon to download its configuration file *.zuliprc* (the structure of this file is explained [here](#configuration-file)
+*  Copy the file to a destination of your choice on your Zulip server, e.g. to `~/.zuliprc`
+5. Subscribe the bot to the streams where you want to use it.
+* By default, the bot *does not intercept/send messages* on any stream.
+* To change that, navigate to *Manage Streams*. Select a stream and add your bot by its email address (the address you assigned in step 2).
+* Now, the bot will do its job on the streams you subscribed it to.
+6. Run the bot.
+*  On your Zulip server, navigate to `~/zulip/contrib_bots/`
+* Run `python run.py ~/zulip/contrib_bots/<my-bot>.py --config-file ~/.zuliprc`
+* Congrats! Now, your bot should be ready to test on the streams you subscribed it to.
+
```

Comment:
Describe what the output looks like when you start a bot.  It shows usage then it shows HTTP connection stuff.